src/Aspire.Dashboard/Components/Dialogs/SettingsDialog.razor.cs

```diff
@@ -10,14 +12,38 @@ namespace Aspire.Dashboard.Components.Dialogs;
 public partial class SettingsDialog : IDialogContentComponent, IDisposable
 {
     private string? _currentSetting;
+    private List<CultureInfo> _languageOptions = null!;
+    private CultureInfo? _selectedUiCulture;
 
     private IDisposable? _themeChangedSubscription;
 
     [Inject]
     public required ThemeManager ThemeManager { get; init; }
 
+    [Inject]
+    public required NavigationManager NavigationManager { get; init; }
+
     protected override void OnInitialized()
     {
+        _languageOptions = DashboardWebApplication.LocalizedCultures
+            .Select(CultureInfo.GetCultureInfo)
+            .OrderBy(c => c.NativeName)
+            .ToList();
+
+        // User may also be using a variant of one of the supported cultures, in which case we should select the culture we support
+        // this doesn't work for zh as we support two different zh cultures
+        if (!_languageOptions.Contains(CultureInfo.CurrentUICulture)
+                 && !StringComparers.Culture.Equals(CultureInfo.CurrentUICulture.TwoLetterISOLanguageName, "zh")
+                 && _languageOptions.FirstOrDefault(c => StringComparers.Culture.Equals(c.TwoLetterISOLanguageName, CultureInfo.CurrentUICulture.TwoLetterISOLanguageName)) is var matchedCulture)
```

Comment:
This seems like it could be prone to mistakes. Tests are needed here. It could be moved to a class and have some unit tests that test different cultures.

I looked at what the middleware does and it uses Parent property on culture. What about using that to match?

https://github.com/dotnet/aspnetcore/blob/81a2bab8704d87d324039b42eb1bab0d977f25b8/src/Middleware/Localization/src/RequestLocalizationMiddleware.cs#L172

Reply:
Thanks, that's a better idea. Updated